cmd/restic/cmd_snapshots.go

```diff
@@ -223,6 +291,40 @@ func PrintSnapshots(stdout io.Writer, list restic.Snapshots, reasons []restic.Ke
 	tab.Write(stdout)
 }
 
+// PrintSnapshotGroupHeader prints which group of the group-by option the
+// following snapshots belong to.
+// Prints nothing, if we did not group at all.
+func PrintSnapshotGroupHeader(stdout io.Writer, groupKeyJSON string, GroupByTag bool, GroupByHost bool, GroupByPath bool) error {
+	if GroupByTag || GroupByHost || GroupByPath {
```

Comment:
It's common in Go to return early, so the actual action in the function is not indented so far. How about this instead?
```suggestion
	if !GroupByTag && !GroupByHost && !GroupByPath {
		return nil
	}
```